packages/sources/vue/src/components/navigation/VtmnSearch/VtmnSearch.vue

```diff
@@ -0,0 +1,99 @@
+<script lang="ts">
+import '@vtmn/css-search/dist/index-with-vars.css';
+import { computed, defineComponent, PropType, reactive } from 'vue';
+import { VtmnSearchVariant, VtmnSearchSize } from './types';
+import { VtmnButton } from '../../index';
+
+export default /*#__PURE__*/ defineComponent({
+  name: 'VtmnSearch',
+  components: { VtmnButton },
+  props: {
+    modelValue: {
+      type: [String, Number] as PropType<string | number>,
+      default: '',
+    },
+    variant: {
+      type: String as PropType<VtmnSearchVariant>,
+      default: 'medium',
+    },
+    disabled: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
+    size: {
+      type: String as PropType<VtmnSearchSize>,
+      default: 'medium',
+    },
+    searchCallback: {
```

Comment:
We can remove this props and avoid to pass a callback function by using instead an emitted event 'search'.

What do you think about it ? It make the component more "stateless".